packages/components/src/components/stepper-horizontal/stepper-horizontal/stepper-horizontal.tsx

```diff
@@ -152,13 +144,16 @@ export class StepperHorizontal {
   };
 
   private defineStepperHorizontalItemElements = (): void => {
-    // TODO: validation? this could be any kind of dom node
+    this.validateComponentChildren();
     this.stepperHorizontalItems = Array.from(this.host.children) as HTMLPStepperHorizontalItemElement[];
   };
 
-  private validateComponent = (): void => {
+  private validateComponentChildren = (): void => {
     throwIfChildrenAreNotOfKind(this.host, 'p-stepper-horizontal-item');
     throwIfChildCountIsExceeded(this.host, 9);
+  };
+
+  private validateComponentState = (): void => {
```

Comment:
Can we combine validations? This wouldn't only make it cleaner but also more precise because when children change their might be potential 2nd current state in the new one?